src/PrestaShopBundle/Resources/config/routing/admin/improve/payment/preferences.yml

```diff
@@ -6,10 +6,34 @@ admin_payment_preferences:
         _legacy_link: AdminPaymentPreferences
         _controller: PrestaShopBundle:Admin\Improve\Payment\PaymentPreferences:index
 
-admin_payment_preferences_process:
-    path: /preferences
+admin_payment_carrier_restrictions_save:
+    path: /carrier-restrictions
     methods: [POST]
     defaults:
+        _controller: 'PrestaShopBundle:Admin\Improve\Payment\PaymentPreferences:processPaymentCarrierRestrictionsForm'
+        _legacy_controller: AdminPaymentPreferences
+        _legacy_link: AdminPaymentPreferences:update
```

Comment:
I'm not sure if it's OK to have duplicate `_legacy_link: AdminPaymentPreferences:update` 😅 

ping @jolelievre

Reply:
You can do it but then it becomes a bit useless because only one will be useful (the first one or last one I don't remember) So you will need a different action to map each of these controllers which is not easy at first sight

But when you look at the initial legacy controller:
https://github.com/PrestaShop/PrestaShop/blob/7940032c451d499b715b170e2670999a38681997/controllers/admin/AdminPaymentPreferencesController.php#L135

You can see that it used special submit parameters for each different actions (mapping different entities or module types) So you can reuse these ones for each individual controller

Besides you can set more than one legacy_link So I would suggest keeping the former common one (`:update`) for each of them and add some new ones For example:

```yaml
admin_payment_carrier_restrictions_save:
        _legacy_link:
            - AdminPaymentPreferences:submitModulereference
            - AdminPaymentPreferences:updateModulereference
            - AdminPaymentPreferences:update

admin_payment_country_restrictions_save:
        _legacy_link:
            - AdminPaymentPreferences:submitModulecountry
            - AdminPaymentPreferences:updateModulecountry
            - AdminPaymentPreferences:update
```